src/ore/src/tracing.rs

```diff
@@ -315,16 +316,51 @@ where
             .with_exporter(exporter)
             .install_batch(opentelemetry::runtime::Tokio)
             .unwrap();
+
+        // By default we turn off tracing from the following crates, because they
+        // have long-lived Spans, which OpenTelemetry does not handle well.
+        //
+        // Note: folks should feel free to add more crates here if we find more
+        // with long lived Spans.
+        let default_targets = [("h2", LevelFilter::OFF), ("hyper", LevelFilter::OFF)];
+
         let (filter, filter_handle) = reload::Layer::new(if otel_config.start_enabled {
-            otel_config.filter
+            let new_targets = Targets::default()
+                .with_targets(default_targets)
+                .with_targets(&otel_config.filter);
+
+            if let Some(default_level) = otel_config.filter.default_level() {
+                new_targets.with_default(default_level)
+            } else {
+                new_targets
+            }
         } else {
             // The default `Targets` has everything disabled.
             Targets::default()
         });
         let layer = tracing_opentelemetry::layer()
+            // OpenTelemetry does not handle long-lived Spans well, and they end up continuously
+            // eating memory until OOM. So we set a max number of events that are allowed to be
+            // logged to a Span, once this max is passed, old events will get dropped
+            //
+            // TODO(parker-timmerman|guswynn): make this configurable with LaunchDarkly
+            .max_events_per_span(2048)
             .with_tracer(tracer)
             .with_filter(filter);
-        let reloader = Arc::new(move |targets| Ok(filter_handle.reload(targets)?));
+        let reloader = Arc::new(move |targets: Targets| {
+            // Re-apply our default targets on reload.
+            let new_targets = Targets::default()
+                .with_targets(default_targets)
+                .with_targets(&targets);
```

Comment:
can you iterate over the `targets` to see if they overrode `h2` or `hyper`, and skip the `default_targets` if so?

Reply:
If the user specified `h2` or `hyper` they should override `default_targets` since we apply `targets` after `default_targets`, does that work? Or would you want to ignore `default_targets` entirely?